src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -25,6 +25,23 @@ namespace RPC {
 
 namespace {
 
+/*  Returns true if the server is stand alone or we think the ledger is
+    historical or we have a recent validated ledger and the ledger's ID is at
+    least 10 larger than the validated. The value 10 is not particularly
+    special here, just some space between the ledgers.
+*/
+bool
+synced (NetworkOPs& netOps, Ledger::pointer ledger)
+{
+    return getConfig ().RUN_STANDALONE ||
```

Comment:
Snark :-) : the logic here is too complex to understand without working out a puzzle.  Imagine how much worse it will get if it needs to be maintained!

Also, what if you need to debug some issue?  There's no way to set breakpoints within a condition to see why it isn't having the desired result.

There are four separate tests in here, struggling to get out... like this:

    // The value 10 is not particularly
    // special here, just some space between the ledgers.
    auto const maxSequenceGap = 10;

    {
        if (getConfig ().RUN_STANDALONE)
            return true;

        if (ledger != netOps.getValidatedLedger () &&
            ledger != netOps.getCurrentLedger () &&
            ledger != netOps.getClosedLedger ())
            return true;

        auto age = getApp ().getLedgerMaster ().getValidatedLedgerAge ();
        if (age > Tuning::maxValidatedLedgerAge)
            return false;

        auto lastSequence = netOps.getValidatedSeq () + maxSequenceGap;
        return ledger->getLedgerSeq () > lastSequence;
    }

I understand that code much better.  I can see where I'd set a breakpoint and examine values to see what was happening.  And it will generate the same code as the "all one expression" solution.